adapters/handlers/rest/handlers_backup_test.go

```diff
@@ -0,0 +1,56 @@
+//                           _       _
+// __      _____  __ ___   ___  __ _| |_ ___
+// \ \ /\ / / _ \/ _` \ \ / / |/ _` | __/ _ \
+//  \ V  V /  __/ (_| |\ V /| | (_| | ||  __/
+//   \_/\_/ \___|\__,_| \_/ |_|\__,_|\__\___|
+//
+//  Copyright © 2016 - 2024 Weaviate B.V. All rights reserved.
+//
+//  CONTACT: hello@weaviate.io
+//
+
+package rest
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/weaviate/weaviate/entities/models"
+	ubak "github.com/weaviate/weaviate/usecases/backup"
+)
+
+func TestCompressionCfg(t *testing.T) {
+	l := "BestSpeed"
+	tcs := map[string]struct {
+		cfg                 *models.BackupConfig
+		expectedCompression ubak.CompressionLevel
+		expectedCPU         int
+		expectedChunkSize   int
+	}{
+		"without config": {
+			cfg:                 nil,
+			expectedCompression: ubak.DefaultCompression,
+			expectedCPU:         ubak.DefaultCPUPercentage,
+			expectedChunkSize:   ubak.DefaultChunkSize,
+		},
+		"with config": {
+			cfg: &models.BackupConfig{
+				CPUPercentage:    25,
+				ChunkSize:        512,
+				CompressionLevel: &l,
```

Comment:
Similar to comment above, what if a user specify just part of the config and not compression Level ? Is this caught in our validation logic or will it fail on ptr dereference ?

Reply:
it has a default, which should be assigned on the validation level